main.rb

```diff
@@ -0,0 +1,229 @@
+      end
+    elsif arg[0]
+      if arg[0] == Regexp
+        my_each do |i|
+          counter = false unless arg[0].match?(i)
+        end
+      elsif arg[0].is_a?(Class)
+        my_each do |i|
+          counter = false unless i.is_a?(arg[0])
+        end
+      else
+        my_each do |i|
+          counter = false unless i == arg[0]
+        end
+      end
+    else
+      my_each do |i|
+        counter = false unless i
+      end
+    end
+    counter
+  end
+
+  def my_any?(*arg)
+    counter = false
+    if block_given?
+      my_each do |i|
+        return true if yield i
+      end
+    elsif arg[0]
+      if arg[0] == Regexp
+        my_each do |i|
+          return true if arg[0].match?(i)
+        end
+      elsif arg[0].is_a?(Class)
+        my_each do |i|
+          return true if i.is_a?(arg[0])
+        end
+      else
+        my_each do |i|
+          return true if i == arg[0]
+        end
+      end
+    else
+      my_each do |i|
+        return true if i
+      end
+    end
+    counter
+  end
+
+  def my_none?(*arg)
+    counter = true
+    if block_given?
+      my_each do |i|
+        counter = false if yield i
+      end
+    elsif arg[0]
+      if arg[0] == Regexp
+        my_each do |i|
+          counter = false if arg[0].match?(i)
+        end
+      elsif arg[0].is_a?(Class)
+        my_each do |i|
+          counter = false if i.is_a?(arg[0])
+        end
+      else
+        my_each do |i|
+          counter = false if i == arg[0]
+        end
+      end
+    else
+      my_each do |i|
+        counter = false if i
+      end
+    end
+    counter
+  end
+
+  def my_count(*arg)
+    counter = 0
+    if block_given?
+      my_each do |i|
+        counter += 1 if yield i
+      end
+    elsif arg[0]
+      if arg[0] == Regexp
+        my_each do |i|
+          counter += 1 if arg[0].match?(i)
+        end
+      elsif arg[0].is_a?(Class)
+        my_each do |i|
+          counter += 1 if i.is_a?(arg[0])
+        end
+      else
+        my_each do |i|
+          counter += 1 if i == arg[0]
+        end
+      end
+    else
+      my_each do |i|
+        counter += 1 if i
+      end
+    end
+    counter
+  end
+
+  def my_map(*arg)
+    map_val = []
+    if block_given? && arg[0].is_a?(Proc)
+      my_each do |i|
+        map_val << i if arg[0].call(i)
+      end
+    elsif block_given?
+      my_each do |i|
+        map_val << i if yield i
+      end
+    elsif arg[0]
+      if arg[0] == Regexp
+        my_each do |i|
+          map_val << i if arg[0].match?(i)
+        end
+      elsif arg[0].is_a?(Class)
+        my_each do |i|
+          map_val << i if i.is_a?(arg[0])
+        end
+      elsif arg[0].is_a?(Proc)
+        my_each do |i|
+          map_val << i if arg[0].call(i)
+        end
+      else
+        my_each do |i|
+          map_val << i if i == arg[0]
+        end
+      end
+    else
+      my_each do |i|
+        map_val << i if i
+      end
+    end
+    map_val
+  end
```

Comment:
# my_map
- [ ] `my_map` returns a new array with the results of running a given block once for every element in enum.
     Failure/Error: expect(array.my_map { |num| num < 10 }).to eq(array.map { |num| num < 10 })

       expected: [true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true...rue, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true]
            got: [1, 0, 3, 6, 8, 1, 8, 6, 3, 7, 5, 6, 2, 1, 0, 0, 4, 4, 3, 4, 4, 2, 3, 1, 6, 5, 8, 6, 3, 8, 1, 6, 0, 8..., 4, 6, 8, 5, 8, 8, 2, 8, 6, 5, 2, 8, 0, 5, 3, 3, 2, 8, 0, 6, 2, 8, 1, 8, 8, 0, 2, 8, 7, 4, 7, 0, 1]

##

- [ ] `my_map` returns a new array with the results of running a given block once for every element in enum.
     Failure/Error: expect(array.my_map(&block)).to eq array.map(&block)

       expected: [true, false, false, false, true, false, false, false, false, false, false, true, true, false, true, ..., true, true, true, true, false, false, false, false, false, false, false, true, false, true, false]
            got: [2, 0, 1, 3, 1, 3, 1, 2, 1, 2, 1, 1, 3, 2, 3, 3, 0, 1, 0, 1, 3, 0, 3, 1, 1, 2, 1, 3, 3, 2, 2, 2, 2, 1, 2, 3, 1, 1, 1, 2, 1, 1, 0, 2]

##    

- [ ] `my_map` returns a new array with the results of running a given block once for every element in enum::range

     Failure/Error: expect(range.my_map(&block)).to eq range.map(&block)

       expected: [false, false, false, false, false, false, false, false, false, false, false, false, false, false, fa...e, false, false, false, false, false, false, false, false, false, false, false, false, false, false]
            got: []

##

- [ ] `my_map` returns an Enumerator if no block is given

     Failure/Error: expect(array.my_map).to be_an(Enumerator)

       expected [8, 3, 7, 2, 5, 1, 5, 2, 7, 3, 8, 6, 4, 2, 5, 0, 3, 1, 0, 7, 2, 7, 2, 0, 7, 1, 8, 7, 1, 3, 3, 0, 7, 8..., 2, 8, 6, 1, 8, 6, 7, 1, 3, 3, 7, 2, 5, 5, 0, 0, 3, 4, 1, 1, 7, 6, 1, 4, 2, 8, 7, 4, 7, 4, 4, 4, 8] to be a kind of Enumerator
##

- [ ] `my_map` accepts a proc as an argument and returns a new array with the results of running the proc once for every element in enum.

      Failure/Error: expect(array.my_map(&my_proc)).to eq array.map(&my_proc)

        expected: [false, false, false, false, false, false, false, false, false, false, false, false, false, false, fa...e, false, false, false, false, false, false, false, false, false, false, false, false, false, false]
             got: []

##
- [ ] `my_map` executes only the proc when both a block and a proc are given and returns a new array with the results of running the proc once for every element in enum.

      Failure/Error: expect(array.my_map(my_proc) { |num| num < 10 }).to eq array.map(&my_proc)

        expected: [false, false, false, false, false, false, false, false, false, false, false, false, false, false, fa...e, false, false, false, false, false, false, false, false, false, false, false, false, false, false]
             got: []